src/server/routes/api/classrooms/index.js

```diff
@@ -1,116 +1,237 @@
 "use strict";
 
 const router = require("express").Router();
-const Classrooms = require("../../../models/classrooms");
 const InviteCodes = require("../../../models/invite-codes");
+const Classrooms = require("../../../models/classrooms");
+const util = require("../../../util");
 
 // returns a list of classrooms this user belongs to
 router.get("/", async (req, res) => {
-  if (req.isAuthenticated()) {
-    Classrooms.find({ $or: [
+  if (!req.isAuthenticated()) {
+    res.status(401).send();
+  }
+
+  try {
+    // get all classrooms with current user
+    // (prepoulated)
+    const classrooms = await Classrooms.find({ $or: [
       { teachers: req.user._id },
       { teacherAssistants: req.user._id },
       { students: req.user._id }
-    ]}).then(classrooms => {
-      res.json(classrooms);
-    }).catch( err => {
-      res.json({ status: false, message: err });
-    });
-  } else {
-    res.status(401).send();
+    ]});
+
+    // remove codes from non-teacher rooms
+    // and cast to object
+    const sanitizedClassrooms = classrooms.map(room => room.sanitize(req.user));
+
+    // send to user
+    return res.json(sanitizedClassrooms);
+  } catch (err) {
+    return res.json({ status: false, message: err.message });
   }
 });
 
 // creates a classroom and returns invite codes for teachers, TAs, and students
 router.post("/", async (req, res) => {
-  if (!req.isAuthenticated()) {
-    return res.status(401).send();
-  }
+  try {
+    if (!req.isAuthenticated()) {
+      return res.status(401).send();
+    }
+
+    // create codes first
+    const codes = await Promise.all([
+        InviteCodes.create({}),
+        InviteCodes.create({}),
+        InviteCodes.create({}),
+    ]);
 
-  return Classrooms.create({
-    schoolId: req.body.schoolId,
-    courseType: req.body.courseType,
-    courseNumber: req.body.courseNumber,
-    sectionNumber: req.body.sectionNumber,
-    courseTitle: req.body.courseTitle,
-    teachers: [req.user._id],
-    teacherAssistants: [],
-    students: []
-  })
-  .then(classroom =>
-      Promise.all([1,2,3].map(n =>
-          InviteCodes.create({ classroom: classroom.id, type: n }))))
-  .then(values => {
-    const [teachers, teacherAssistants, students] = values.map(invite => invite.code);
-    res.json({ teachers, teacherAssistants, students });
-  })
-  .catch(err => res.json({ status: false, message: err }));
+    // create classroom with codes
+    return Classrooms.create({
+      schoolId: req.body.schoolId,
+      courseType: req.body.courseType,
+      courseNumber: req.body.courseNumber,
+      sectionNumber: req.body.sectionNumber,
+      courseTitle: req.body.courseTitle,
+      teachers: [req.user._id],
+      teacherAssistants: [],
+      students: [],
+      teacherCode: codes[0].code,
+      taCode: codes[1].code,
+      studentCode: codes[2].code,
+    })
+    .then( c => res.json(c.sanitize(req.user)))
```

Comment:
putting `.then` and `.catch` inside try/catch blocks is an anti-pattern. It prevents the try/catch block from functioning properly. See pull request #22